src/MergeContext.js

```diff
@@ -954,12 +962,50 @@ class PullRequest {
         return false;
     }
 
+    async _getMergeCommit() {
+        const mergeSha = await GH.getReference("pull/" + this._prNumber() + "/merge");
+        return await GH.getCommit(mergeSha);
+    }
+
+    // obtains abandoned staging checks
+    async _getAbandonedStagingStatuses() {
+        assert(!this._stagedSha());
+
+        const allEvents = await GH.getIssueEvents(this._prNumber());
+        // staging events are PR events where the bot user created a commit referencing this PR
+        let stagingEvents = allEvents.filter(ev => ev.event === "referenced" && ev.actor.login === Config.githubUserLogin());
+        if (!stagingEvents.length)
+            return;
+
+        // just in case: events should be already in chronological order already
+        stagingEvents = stagingEvents.sort((ev1, ev2) => Date.parse(ev1.created_at) - Date.parse(ev2.created_at));
+        const lastStagingEvent = stagingEvents[stagingEvents.length - 1];
+        const mergeCommit = await this._getMergeCommit();
+        const mergeCommitCreatedAt = Date.parse(mergeCommit.author.date); // for merge commits author.date and committer.date should be the same
+        const lastStagingEventCreatedAt = Date.parse(lastStagingEvent.created_at);
+        assert(mergeCommitCreatedAt);
+        assert(lastStagingEventCreatedAt);
+        // the staging event for merge commit X is always created after X so we do not
+        // need to deal with the lastStagingEventCreatedAt == mergeCommitCreatedAt case
+        assert(lastStagingEventCreatedAt != mergeCommitCreatedAt);
+        if (lastStagingEventCreatedAt < mergeCommitCreatedAt)
+            return; // merge commit created after the last staging event
+
+        // Merge commit created before the last staging event.
+        // Thus, we can get statuses of that event commit SHA.
```

Comment:
Should we say "we can use statuses..." instead of "we can get statuses..."? We can get statuses of any tested commit, right? We just do not want use the old/irrelevant statuses, right?

Reply:
You are right that we can get statuses of any commit - as far as we have its SHA. Probably we can drop this comment at all - especially if we start comparing tree SHAs, as I suggested above.